bin/radical-pilot-service-wrapper

```diff
@@ -0,0 +1,265 @@
+#
+# This script is a wrapper around an actual executable.  It's purpose is,
+# depening on the mode of operation, to either watch the executable's stdout for
+# some pattern, or to watch an output file created by the executable for that
+# pattern.  If the pattern is found, the matching data is extracted and sent to
+# *this* script's stdout.  If the executable returns a non-zero exit code, or if
+# a timeout is reached, the script will also return a non-zero exit code.  If
+# The script returns a zero exit code but a pattern is provided but not found,
+# the script will also return a non-zero exit code.
+#
+# The script is called with the following arguments:
+#
+#   -m mode   : data stream to search through, either `stdout` (default),
+#               `stderr`, or a file name
+#   -p pattern: regex pattern to search for
+#   -o output : file name to store the executable's stdout
+#   -e error  : file name to store the executable's stderr
+#   -t timeout: time in seconds after which the script will terminate if pattern
+#               is not found
+#   --        : marks the end of the script's arguments and the start of the
+#               executable's name and arguments
+
+
+CHANNEL  = 'control_pubsub'
+last_msg = None
+
+
+# ------------------------------------------------------------------------------
+#
+def main():
+
+    global last_msg
+
+    print('starting')
+
+    parser = argparse.ArgumentParser(description='')
+
+    parser.add_argument('-c', '--command',  dest='command')
+    parser.add_argument('-r', '--registry', dest='reg_url')
+    parser.add_argument('-u', '--uid',      dest='uid')
+    parser.add_argument('-m', '--mode',     dest='mode')
+    parser.add_argument('-p', '--pattern',  dest='pattern')
+    parser.add_argument('-o', '--output',   dest='output')
+    parser.add_argument('-e', '--error',    dest='error')
+    parser.add_argument('-v', '--verbose',  dest='verbose', action='store_true')
+    parser.add_argument('-t', '--timeout',  dest='timeout')
+
+    args    = parser.parse_args()
+    command = args.command or None
+    reg_url = args.reg_url or os.environ.get('RP_REGISTRY_ADDRESS')
+    uid     = args.uid     or os.environ.get('RP_TASK_ID')
+    mode    = args.mode    or 'stdout'
+    pattern = args.pattern or '\\<http[s]?://[0-9a-zA-Z\\.:_-]*\\>'
+    output  = args.output  or '%s.stdout' % uid
+    error   = args.error   or '%s.stderr' % uid
+    verbose = args.verbose or False
+    timeout = args.timeout or 0
+
+    ranks   = int(os.environ.get('RP_RANKS'))
+    rank    = int(os.environ.get('RP_RANK'))
+
+    output  = '%s.%04d' % (output, rank)
+    error   = '%s.%04d' % (error,  rank)
+
+    timeout = int(timeout) if timeout else 0
+
+    if not command: raise ValueError('no command given')
+    if not uid    : raise ValueError('no uid given')
+    if not reg_url: raise ValueError('RP_REGISTRY_ADDRESS not set')
+
+    # get control pubsub from registry
+    print('reg_url x: %s' % reg_url)
```

Comment:
we shouldn't use service output file for RP related messages, should we have a dedicated log file?

Reply:
Aaactually - it's a debug message, but the service writes to a different stdout, so this should be fine?

Reply:
This is resolved now, the wrapper writes a separate log file.